src/sentry/release_health/metrics_sessions_v2.py

```diff
@@ -0,0 +1,595 @@
+""" This module offers the same functionality as sessions_v2, but pulls its data
+from the `metrics` dataset instead of `sessions`.
+
+Do not call this module directly. Use the `release_health` service instead. """
+import abc
+from collections import defaultdict
+from dataclasses import dataclass, replace
+from datetime import datetime
+from typing import (
+    Any,
+    Generator,
+    List,
+    Mapping,
+    MutableMapping,
+    Optional,
+    Sequence,
+    Set,
+    Tuple,
+    Union,
+    cast,
+)
+
+from snuba_sdk import Column, Condition, Entity, Op, Query
+from snuba_sdk.expressions import Granularity
+from snuba_sdk.function import Function
+from snuba_sdk.legacy import json_to_snql
+from typing_extensions import Literal, TypedDict
+
+from sentry.release_health.base import (
+    GroupByFieldName,
+    SessionsQueryFunction,
+    SessionsQueryGroup,
+    SessionsQueryResult,
+    SessionsQueryValue,
+)
+from sentry.sentry_metrics import indexer
+from sentry.snuba.dataset import Dataset, EntityKey
+from sentry.snuba.metrics import TS_COL_GROUP, TS_COL_QUERY, get_intervals
+from sentry.snuba.sessions_v2 import QueryDefinition, finite_or_none
+from sentry.utils.snuba import raw_snql_query
+
+
+def _resolve(org_id: int, name: str) -> Optional[int]:
+    """Wrapper for typing"""
+    return indexer.resolve(org_id, name)  # type: ignore
+
+
+def _resolve_ensured(org_id: int, name: str) -> int:
+    """Assume the index entry exists"""
+    index = _resolve(org_id, name)
+    assert index is not None
+    return index
+
+
+def _reverse_resolve(org_id: int, index: int) -> Optional[str]:
+    """Wrapper for typing"""
+    return indexer.reverse_resolve(org_id, index)  # type: ignore
+
+
+def _reverse_resolve_ensured(org_id: int, index: int) -> str:
+    """Assume the index entry exists"""
+    string = _reverse_resolve(org_id, index)
+    assert string is not None
+    return string
+
+
+_SessionStatus = Literal[
+    "abnormal",
+    "crashed",
+    "errored",
+    "healthy",
+]
+
+
+@dataclass(frozen=True)
+class _StatusValue:
+    """A data value associated with a certain session status"""
+
+    session_status: Optional[_SessionStatus]
+    value: Union[None, float, int]
+
+
+_MetricName = Literal["session", "session.duration", "session.error", "user"]
+
+
+#: Name of a column returned by snuba
+_Column = Literal["value", "avg", "max", "p50", "p75", "p90", "p95", "p99"]
+
```

Comment:
A bit confusing in my opinion, shouldn't this be _ColumnName or even _SnubaColumnName (since we already have a `Column` type which represents a column)

Reply:
Good point, hope that my latest commit clarifies.